Libraries/lwIP/lwip.mk

```diff
@@ -30,32 +30,38 @@
  # ownership rights.
  #
  ###############################################################################
- 
-ifeq "$(LWIP_DIR)" ""
-$(error LWIP_DIR must be specified")
-endif
-
-TARGET_UC:=$(shell echo $(TARGET) | tr a-z A-Z)
-TARGET_LC:=$(shell echo $(TARGET) | tr A-Z a-z)
 
-# Specify the library variant.
-ifeq "$(MFLOAT_ABI)" "hardfp"
-LIBRARY_VARIANT=hardfp
-else
-ifeq "$(MFLOAT_ABI)" "hard"
-LIBRARY_VARIANT=hardfp
+# Specify the build directory if not defined by the project
+ifeq "$(BUILD_DIR)" ""
+ifeq "$(MFLOAT_ABI)" ""
+LWIP_BUILD_DIR = ${LWIP_DIR}/bin
 else
-LIBRARY_VARIANT=softfp
+LWIP_BUILD_DIR = ${LWIP_DIR}/bin/$(MFLOAT_ABI)
```

Comment:
What do you think about making default binary location to project's build directory?
```
LWIP_BUILD_DIR=$(CURDIR)/build/lwIP
```
This may also overcome some issues about library build configuration differences. You can still keep prebuild library as it is.

Reply:
My only hesitation for doing this is that it's a somewhat larger library.  Since users will not typically modify the lwIP source code, it will affect build times.

I think there are some other solutions we can try to address the pre-built libraries in a future PR.